src/catchup/simulation/TxSimApplyTransactionsWork.cpp

```diff
@@ -398,30 +398,38 @@ TxSimApplyTransactionsWork::scaleLedger(
 bool
 TxSimApplyTransactionsWork::getNextLedgerFromHistoryArchive()
 {
-    if (mStream->getNextLedger(mHeaderHistory, mTransactionHistory,
-                               mResultHistory))
+    try
     {
-        // Derive transaction apply order from the results
-        UnorderedMap<Hash, TransactionEnvelope> transactions;
-        for (auto const& tx : mTransactionHistory.txSet.txs)
+        if (!mStream->getNextLedger(mHeaderHistory, mTransactionHistory,
+                                    mResultHistory))
         {
-            auto txFrame = TransactionFrameBase::makeTransactionFromWire(
-                mApp.getNetworkID(), tx);
-            transactions[txFrame->getContentsHash()] = tx;
+            return false;
         }
+    }
+    catch (xdr::xdr_bad_message_size&)
+    {
+        LOG_ERROR(DEFAULT_LOG, "Failed to read xdr");
```

Comment:
similarly here, since we couldn't read the next ledger, `mHeaderHistory` is now invalid, so we need to exit the simulation gracefully. I think the simplest way to do it is to catch the exception here and return `WORK_FAILURE`:
https://github.com/stellar/stellar-core/blob/bdb165b75613bf23e638dd32f8a6c9c07f26791f/src/catchup/simulation/TxSimApplyTransactionsWork.cpp#L578 For clarity, maybe also include a log hinting that there is potential version problem

Reply:
Fixed